feeds/scheduler/feed_group.go

```diff
@@ -31,28 +43,36 @@ func (fg *FeedGroup) AddFeed(feed feeds.ScheduledFeed) {
 }
 
 func (fg *FeedGroup) Run() {
-	_, err := fg.PollAndPublish()
-	if err != nil {
-		log.Error(err)
+	result := fg.pollAndPublish()
+	if result.pollErr != nil {
+		log.Error(result.pollErr)
+	}
+	if result.pubErr != nil {
+		log.Error(result.pubErr)
 	}
 }
 
-func (fg *FeedGroup) PollAndPublish() (int, error) {
+func (fg *FeedGroup) pollAndPublish() groupResult {
+	result := groupResult{}
 	pkgs, errs := fg.poll()
-	if len(errs) > 0 {
-		return 0, errs[0]
-	} else if len(pkgs) == 0 {
-		return 0, nil
-	}
 
+	if len(errs) != 0 {
```

Comment:
Accidentally deleted the previous comment... We should remove the logging [here](https://github.com/ossf/package-feeds/pull/123/files#diff-95d0a7cd43229270ff6fe6008662c35352c7819fe8be4f7e313e5b5357281ba0R97) and move it to a higher level to avoid returning errors which have already been logged.

The error list can be handled without the loop via `errs = append(errs, result.errs...)`

Reply:
I've updated this such that `poll()` still logs the errors, but now it only returns `errPoll/nil`. Keeping the errors logged within poll allows them to be easily logged with the feedname fields, without further wrapping. As a followup, when tackling #101 `publishPackages` should also be modified to log the errors and only return `errPub/nil'